### ISSUE TITLE
Concept: Clusters as sources

### ISSUE THREAD

**Issue by martindurant** [intake/intake#535]
To have a catalogue where the entries are your clusters. Could use this as a mechanism to store credentials/endpoints or even to spin up new clusters. A `to_dask` method might, for instance, give you back the cluster and/or client object.

(@mrocklin : this was suggested in our monthly meeting as an extension of the catalogue concept, where Intake could hook into whatever config system Coiled or other providers use)

**Comment by mrocklin** [comment]
Yeah, that could be fun.  I imagine that there are a few groups that would
want to switch between on-prem and cloud, or cpu and gpu configurations.
